src/utils/PreProcessQueue.js

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable consistent-return */
+
+import Queue from '@segment/localstorage-retry';
+
+const queueOptions = {
+  maxRetryDelay: 360000,
+  minRetryDelay: 1000,
+  backoffFactor: 2,
+  maxAttempts: Infinity,
+  maxItems: 100,
+};
+
+class PreProcessQueue {
+  constructor() {
+    this.data = undefined;
+    this.callback = undefined;
+  }
+
+  init(options, callback) {
+    if (options) {
+      // TODO: add checks for value - has to be +ve?
+      Object.assign(queueOptions, options);
+    }
+    if (callback) {
+      this.callback = callback;
+    }
+    this.payloadQueue = new Queue('rs_events', queueOptions, (item, done) => {
+      const { type, rudderElement } = item;
+      this.processQueueElement(type, rudderElement, (err, res) => {
+        if (err) {
+          return done(err);
+        }
+        done(null, res);
+      });
+    });
+    this.payloadQueue.start();
+  }
+
+  setCloudModeEventsIntegrationObjData(integrationsData) {
+    // An indicator to process elements in queue
+    this.data = integrationsData;
+  }
+
+  processQueueElement(type, rudderElement, queueFn) {
+    try {
+      if (this.data) {
+        // if not specified at event level, All: true is default
+        const clientSuppliedIntegrations = rudderElement.message.integrations || { All: true };
```

Comment:
What is the need to do this operation here only?

Reply:
removed this line to `processQueueElement` function as it's same for both